Makefile

```diff
@@ -0,0 +1,27 @@
+TSX_FILES = $(shell find code -type f -name '*.tsx')
+JS_FILES = $(patsubst code/%.tsx, artifacts/code/%.js, $(TSX_FILES))
+.PHONY: all
+all: node_modules artifacts/code/index.html $(JS_FILES)
+
+clean:
+	rm -rf artifacts
+
+very-clean:
+	rm -rf artifacts/code/* typings node_modules
+
+typings: typings.json
+	typings install
+
+artifacts/code/%.js: code/%.tsx typings
+	@mkdir -p "$(@D)"
+	tsc \
```

Comment:
What do you think about using a `tsconfig.json` file for holding `tsc` parameters? Easier to change a JSON file than a Makefile. `tsc` will automatically use a `tsconfig.json` file if it's in the `pwd`